omegaconf/omegaconf.py

```diff
@@ -739,7 +744,8 @@ def update(
         :param merge: If value is a dict or a list, True for merge, False for set.
             True to merge
             False to set
-            None (default) : deprecation warning and default to False
+            None (default): deprecation warning and default to False
+        :param force_add: insert the entire path regardless of Struct flag or Structured Config nodes.
```

Comment:
It may be surprising for `force_add` not to allow adding to readonly configs, is that intended?

(if it's changed, it may be easier to rename it into `force` so that it can force both adding new values and overriding existing ones, as otherwise you'd need to make the difference between both situations for readonly configs, which seems tricky)

Reply:
The [reported problem](https://github.com/facebookresearch/hydra/issues/1515) is specifically about adding a field to a Structured Config.
The original implementation in Hydra was using `open_dict on the root of the config and this was not sufficient because of Structured Configs are not inheriting the struct flag from their parent.

read-only is used as a mechanism to protect against changes to nodes [here](https://hydra.cc/docs/patterns/write_protect_config_node).
In contrast to the reported problem, using `with read_write(cfg):` will work fine because it's behavior normal and in fact frozen support for dataclasses is implemented by setting the recursive read-only flag on the DictConfig node.

Reply:
I see. I feel like it may be possible to have a version of `open_dict()` (or, more generally, `flag_override()`) that behaves as if *all* children had their flag set to the desired value (which would avoid the issue of intermediate nodes causing trouble because of their local flags). It'd be a bit tricky to get it to work efficiently though (you don't want to actually propagate the desired flag to all children -- only those whose flags are being accessed within the context)...